homeassistant/components/mawaqit/__init__.py

```diff
@@ -0,0 +1,501 @@
+
+from collections.abc import Callable
+from datetime import datetime, timedelta
+import json
+import logging
+import os
+import shutil
+from typing import Any
+
+from dateutil import parser as date_parser
+from mawaqit.consts import BadCredentialsException
+from requests.exceptions import ConnectionError as ConnError
+import voluptuous as vol
+
+from homeassistant.config_entries import SOURCE_IMPORT, ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers.dispatcher import async_dispatcher_send
+from homeassistant.helpers.event import (
+    async_call_later,
+    async_track_point_in_time,
+    async_track_time_change,
+)
+from homeassistant.helpers.storage import Store
+from homeassistant.helpers.typing import ConfigType
+import homeassistant.util.dt as dt_util
+from homeassistant.util.dt import now as ha_now
+
+from . import utils
+from .const import (
+    CONF_CALC_METHOD,
+    DATA_UPDATED,
+    DEFAULT_CALC_METHOD,
+    DOMAIN,
+    MAWAQIT_ALL_MOSQUES_NN,
+    MAWAQIT_MY_MOSQUE_NN,
+    MAWAQIT_PRAY_TIME,
+    MAWAQIT_STORAGE_KEY,
+    MAWAQIT_STORAGE_VERSION,
+    UPDATE_TIME,
+)
+
+CURRENT_DIR = os.path.dirname(os.path.realpath(__file__))
+file_path = f"{CURRENT_DIR}/data/mosq_list_data"
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORMS = ["sensor"]
+
+
+try:
+    with open(file_path, encoding="utf-8") as text_file:
+        data = json.load(text_file)
+
+    # Accessing the CALC_METHODS object
+    CALC_METHODS = data["CALC_METHODS"]
+except FileNotFoundError:
+    # First Run
+    _LOGGER.warning("The file %s was not found", file_path)
+    CALC_METHODS = []
+
+
+CONFIG_SCHEMA = vol.Schema(
+    vol.All(
+        cv.deprecated(DOMAIN),
+        {
+            DOMAIN: {
+                vol.Optional(CONF_CALC_METHOD, default=DEFAULT_CALC_METHOD): vol.In(
+                    CALC_METHODS
+                ),
+            }
+        },
+    ),
+    extra=vol.ALLOW_EXTRA,
+)
+
+
+def is_date_parsing(date_str):
+    """Check if the given string can be parsed into a date."""
+    try:
+        return bool(date_parser.parse(date_str))
+    except ValueError:
+        return False
+
+
+async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
+    """Import the Mawaqit Prayer component from config."""
+    if DOMAIN in config:
+        hass.async_create_task(
+            hass.config_entries.flow.async_init(
+                DOMAIN, context={"source": SOURCE_IMPORT}, data=config[DOMAIN]
+            )
+        )
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> bool:
+    """Set up the Mawaqit Prayer Component."""
+
+    hass.data.setdefault(DOMAIN, {})
+    client = MawaqitPrayerClient(hass, config_entry)
+
+    if not await client.async_setup():
+        return False
+
+    hass.data[DOMAIN] = client
```

Comment:
```suggestion
    config_entry.runtime_data = client
```

Please update the type of the config entry and the usage throughout the PR